tests/production_restart.py

```diff
@@ -64,28 +64,54 @@
     cluster.biosNode.kill(signal.SIGTERM)
     cluster.waitOnClusterSync(blockAdvancing=5)
 
-    node0 = cluster.getNode(0) # producer
-    node1 = cluster.getNode(1) # finalizer
+    producerNode = cluster.getNode(0)
+    finalizerNode = cluster.getNode(1)
 
     Print("Wait for lib to advance")
-    assert node1.waitForLibToAdvance(), "Node1 did not advance LIB"
-    assert node0.waitForLibToAdvance(), "Node0 did not advance LIB"
+    assert finalizerNode.waitForLibToAdvance(), "finalizerNode did not advance LIB"
+    assert producerNode.waitForLibToAdvance(), "producerNode did not advance LIB"
 
     Print("Set finalizers so a pending is in play")
-    assert cluster.setFinalizers([node1, node0], node0), "setfinalizers failed" # switch order
-    assert node0.waitForLibToAdvance(), "Node0 did not advance LIB after setfinalizers"
-    node0.waitForHeadToAdvance() # get additional qc
-
-    Print("Shutdown producer node0")
-    node0.kill(signal.SIGTERM)
-    assert not node0.verifyAlive(), "Node0 did not shutdown"
-
-    Print("Restart producer node0")
-    node0.relaunch(chainArg=" -e ")
+    # Try a number of times to make sure we have a pending policy
+    numTrys = 10
+    for i in range(0, numTrys):
+       # Switch BLS keys on producerNode. setFinalizers uses the first key
+       # in the configured key list (index 0)
+       newBlsPubKey = producerNode.keys[1].blspubkey
+       newBlsPrivKey = producerNode.keys[1].blsprivkey
+       newBlsPop = producerNode.keys[1].blspop
+       producerNode.keys[1].blspubkey = producerNode.keys[0].blspubkey
+       producerNode.keys[1].blsprivkey = producerNode.keys[0].blsprivkey
+       producerNode.keys[1].blspop = producerNode.keys[0].blspop
+       producerNode.keys[0].blspubkey = producerNode.keys[1].blspubkey
+       producerNode.keys[0].blsprivkey = producerNode.keys[1].blsprivkey
+       producerNode.keys[0].blspop = producerNode.keys[1].blspop
```

Comment:
This doesn't really switch keys, right? I think you meant to use `newBls...`.

Reply:
My bad. Thanks!